src/main/scala/com/thenewmotion/chargenetwork/ocpp/charger/BosService.scala

```diff
@@ -109,7 +129,30 @@ class ConnectorServiceImpl(protected val service: SyncCentralSystem, connectorId
     service(MeterValuesReq(ConnectorScope(connectorId), Some(transactionId), List(meter)))
   }
 
+  def signedAblMeterValue(transactionId: Int, meterValue: Int): Unit = {
+    service(MeterValuesReq(ConnectorScope(connectorId), Some(transactionId), List(signedMeterValuesBeginTransaction(meterValue))))
+  }
+
+  def signedMeterValuesBeginTransaction(meter: Int): Meter = {
+    val meterValue: Value = Value(value = meter.toString,
+      context = ReadingContext.TransactionBegin,
+      format = ValueFormat.Raw,
+      measurand = Measurand.EnergyActiveImportRegister,
+      phase = None,
+      location = Location.Outlet,
+      unit = UnitOfMeasure.Wh)
+
+    val signedMeterReading: Value = meterValue.copy(context = ReadingContext.TransactionBegin,
+      format = ValueFormat.SignedData, value = signedMeterValue)
+
```

Comment:
I see that here you are copying the original and overwriting some of the fields. In the case where the format is a SIGNED many of those fields are redundant and should not be specified, such as phase, location etc. They are relevant to the meter reading itself, not the signature.

Reply:
All of those fields are required by the constructor of `Value` class. Even if I don't clone the object overriding certain data I would have to declare all of those fields in order to instantiate  `signedMeterReading `. The only optional field in this class is `phase` and that's also set to `None`

Reply:
OK, clear